src/main/java/com/fasterxml/jackson/databind/DefaultTyping.java

```diff
@@ -40,13 +40,19 @@
      * "natural" types (String, Boolean, Integer, Double), which
      * can be correctly inferred from JSON; as well as for
      * all arrays of non-final types.
+     * Even if the type is final, when serialized as a root node (not as a field),
```

Comment:
No, I don't think this is true. Do you have a test showing that this would occur? Specific check is:

```
            case NON_FINAL:
                while (t.isArrayType()) {
                    t = t.getContentType();
                }
                while (t.isReferenceType()) {
                    t = t.getReferencedType();
                }
                return !t.isFinal() && !TreeNode.class.isAssignableFrom(t.getRawClass());```
```

and should be applied for root values too, as far as I know (but I could be wrong).
I would like to verify this because if check was not applied to root values then quite a bit of other documentation would need to change too.

Reply:
Check it again, you are right. There is a bug in my unit tests, my fault.

Reply:
Check it again, you are right. There is a bug in my unit tests, my fault.
And I will create another PR for annotation soon once I have a moment.

Reply:
Ah ok good -- thank you for checking, I just wanted to make sure I did not miss something -- many times I do find bugs this way.